recipe/meta.yaml

```diff
@@ -0,0 +1,143 @@
+{% set version = "2.1.2" %}
+{% set run_unittests = False %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/audio/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 82c2745a73172b495327ec36c6970ad5ad9d5d2ac44feeaea2617152f9393bf7
+
+build:
+  number: 0
+  skip: True  # [py>311]
+  script:
+    - export USE_FFMPEG=ON                         # [not win]
+    - set USE_FFMPEG=ON                            # [win]
+    - export BUILD_SOX=OFF                         # [not win]
+    - set BUILD_SOX=OFF                            # [win]
+    - export USE_OPENMP=ON                         # [not win]
+    - set USE_OPENMP=ON                            # [win]
+    - export BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON  # [not win]
+    - set BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON     # [win]
+    - export FFMPEG_ROOT=${PREFIX}                 # [not win]
+    - set FFMPEG_ROOT=%LIBRARY_PREFIX%             # [win]
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - make  # [not win]
+    - patchelf  # [linux]
+    - protobuf {{ libprotobuf }}
+    - llvm-openmp  # [osx]
+    - intel-openmp  # [not osx and blas_impl == 'mkl']
+  host:
+    - python
+    - setuptools
+    - wheel
+    - pip
+    - pytorch 2.1.0
+    - pybind11
+    - ffmpeg >=6,<7
+  run:
+    - python
+    # according to compatibility matrix in
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#compatibility-matrix
+    - pytorch >=2.1.0,<2.2.0
+    # according to optional dependencies
+    # https://github.com/pytorch/audio/blob/v2.1.2/docs/source/installation.rst#optional-dependencies
+    # in this special case we want users of channel sfe140ed to be able to use the load and save functions, therefore
+    # at least ffmpeg (the mightiest of the backends) is specified as a run dependency.
+    - ffmpeg >=6,<7
+    # this is not officially required but it is suggested in some upstream build examples and allows the correct library to be used 
+    #- libprotobuf <4.21.0
+
+# ignoring sox tests because we don't integrate sox
+{% set ignore_modules =                  " --ignore=test/torchaudio_unittest/backend/soundfile --ignore=test/torchaudio_unittest/backend/dispatcher/soundfile" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/torchaudio_unittest/backend/dispatcher/sox --ignore=test/torchaudio_unittest/backend/sox_io" %}
+# these tests import torchaudio.prototype which does not exist in main module
+{% set ignore_modules = ignore_modules + " --ignore=test/integration_tests/prototype --ignore=test/integration_tests/rnnt_pipeline_test.py" %}
+
+# test to skip because of missing deep-phonemizer
+{% set ignore_modules = ignore_modules + " --ignore=test/integration_tests/ctc_decoder_integration_test.py" %}
+# test to skip because of missing dependency
```

Comment:
a bit more detail to these comments would be helpful:
- which missing dependency? (why can't you add the dependency?)
- what kind of URL errors and why only osx-64? seems like a "URL error" would affect osx-arm64 as well? (I think they're both physical builders in the same building, could be wrong)
- which unsupported torchaudio function and why only osx-64?
- what unexpected argument to load() and why shouldn't it be solved with a patch?
also would be good to be sure the segfault mentioned has gone away by running all tests and or-truing the call to pytest so the tests run through failures

Reply:
1. The URL errors I only got recently on `osx-64`. I think it's safe to assume that they were affected by some network issue since I was able to run them before.

2. I would really not spend time on patching test because of some unexpected argument to the load function. These are few tests among several thousands. Also, patching the upstream code doesn't seem like a good idea either: I did some research on the issue and found no mentions. It's more likely that the tests are not up to date. If I manage to find the logs for those failed tests, I will further document them.

Reply:
4. which unsupported torchaudio function and why only osx-64? I will try to document this further.

Reply:
1. This good info - basically "URL error" kind of suggests an error in the URL, i.e. an incorrect URL in the test code, so it might be better to re-phrase
2. again, this is good info. In principle we need to know that the issue is in the test not the main code
3. great, thanks
4. I'm referrring to the comment `tests to ignore because of unsupported torchaudio function`. Which torchaudio function are you referring to? and we need to know that the error is in the test not because we have left something out of the configuration.

Reply:
It seems I cannot reproduce the `unsupported torchaudio function` problem 😕 
I removed the skipped tests and I'm testing them on `osx-arm64` and `linux-64` (as we agreed in the past that the unit tests tend to take too many resources from our builders)

Reply:
I also fixed the URL error comment. commit being pushed soon.